src/tree/AttachedAccountsTreeItem.ts

```diff
@@ -81,21 +83,73 @@ export class AttachedAccountsTreeItem implements IAzureParentTreeItem {
 
             if (connectionString) {
                 let treeItem: IAzureTreeItem = await this.createTreeItem(connectionString, defaultExperience);
-                if (this._attachedAccounts.find(s => s.id === treeItem.id)) {
-                    vscode.window.showWarningMessage(`Database Account '${treeItem.id}' is already attached.`)
-                } else {
-                    this._attachedAccounts.push(treeItem);
-                    if (this._keytar) {
-                        await this._keytar.setPassword(this._serviceName, treeItem.id, connectionString);
-                        await this.persistIds();
-                    }
-                }
+                await this.attachAccount(treeItem, connectionString);
             }
         } else {
             throw new UserCancelledError();
         }
     }
 
+    public async attachEmulator(): Promise<void> {
+        let connectionString: string;
+        const defaultExperience = <Experience>await vscode.window.showQuickPick(['MongoDB', 'DocumentDB'], { placeHolder: "Select a Database Account API...", ignoreFocusOut: true });
+        let validateInput: (input: string) => string | undefined | null = (input: string) => {
+            try {
+                if (parseInt(input).toString() === input) {
+                    return;
+                }
+                else {
+                    return "Port must be finite integer";
+                }
+            }
+            catch {
+                return "Port must be finite integer";
+            }
+        };
+
+        if (defaultExperience) {
+            let defaultPort: string;
+            if (defaultExperience === Experience.MongoDB) {
+                defaultPort = "10255";
+            }
+            else {
+                defaultPort = "8081";
+            }
+            const port: string = await vscode.window.showInputBox({
+                value: defaultPort,
+                prompt: 'Enter the localhost port for the emulator',
+                validateInput: validateInput,
+                ignoreFocusOut: true
+            });
```

Comment:
Allow entering a port for this: 
https://stackoverflow.com/questions/44851311/cosmosdb-emulator-cant-start-since-port-is-already-in-use

Reply:
I agree more with the vscode setting. The default attach will have the same issue of having to identify when it's an emulator to disable SSL-checks.